recipes/shogun-cpp/meta.yaml

```diff
@@ -0,0 +1,79 @@
+{% set name = "shogun-cpp" %}
+{% set version = "6.0.0" %}
+{% set sha256 = "413da26db7fb57aa36c2981bf92d23817c0a29f22051b331dbe5b86e46119748" %}
+{% set blas_variant = "openblas" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: shogun_{{ version }}.tar.gz
+  url: https://github.com/shogun-toolbox/shogun/archive/shogun_{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - use-detected-lapack.patch  # From upstream: properly link to OpenBLAS
+
+build:
+  number: 0
+  features:
+    - blas_{{ blas_variant }}
+  skip:
+    - True  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - arpack
+    - bzip2
+    - cmake
+    - ctags
+    - curl
+    - eigen
+    - glpk
+    - hdf5
+    - json-c
+    - libxml2
+    - lzo
+    - openblas
+    - protobuf
+    - snappy
+    - xz
+    - zlib
```

Comment:
Okay, I took reasonable guesses at the needed versions for those (based on what's in conda-forge + abi-laboratory.pro) and pinned them.